File: Data Structures & Algorithms/maximum-subarray/submission-0.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int n = nums.size(),ans = nums[0],prev = nums[0] ;
        for(int i=1;i<n;i++){
            prev = max(prev + nums[i],nums[i]) ;
            ans = max(ans,prev) ;
        }
        return ans ;
    }
};

File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        l,r,ans = 0,len(heights)-1,0
        while r > l:
            area = abs(r-l) * min(heights[l],heights[r]) 
            ans = max(ans,area) 
            if heights[l] <= heights[r]:
                l += 1 
            else:
                r -= 1
        
        return ans